src/spoke/VaultSpoke.sol

```diff
@@ -0,0 +1,381 @@
+    bytes calldata signature
+  ) public returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.depositor, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.depositor, params.nonce);
+    return
+      _executeDeposit({
+        depositor: params.depositor,
+        receiver: params.receiver,
+        assets: params.assets
+      });
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function mintWithSig(
+    EIP712Types.VaultMint calldata params,
+    bytes calldata signature
+  ) public returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.depositor, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.depositor, params.nonce);
+    return
+      _executeMint({depositor: params.depositor, receiver: params.receiver, shares: params.shares});
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function withdrawWithSig(
+    EIP712Types.VaultWithdraw calldata params,
+    bytes calldata signature
+  ) public returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+    return
+      _executeWithdraw({
+        caller: msg.sender,
+        receiver: params.receiver,
+        owner: params.owner,
+        assets: params.assets
+      });
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function redeemWithSig(
+    EIP712Types.VaultRedeem calldata params,
+    bytes memory signature
+  ) public returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+    return
+      _executeRedeem({
+        caller: msg.sender,
+        receiver: params.receiver,
+        owner: params.owner,
+        shares: params.shares
+      });
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function depositWithPermit(
+    uint256 assets,
+    address receiver,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external returns (uint256) {
+    try
+      IERC20Permit(asset()).permit({
+        owner: msg.sender, // deposit only mints for caller
+        spender: address(this),
+        value: assets,
+        deadline: deadline,
+        v: v,
+        r: r,
+        s: s
+      })
+    {} catch {}
+    return deposit(assets, receiver);
+  }
+
+  /// @inheritdoc IERC20Permit
+  function permit(
```

Comment:
reviewer note: this is permit for vault share token, `depositWithPermit` consumes permit for token
also   // we re-reimplement permit & `nonces(address)` instead of inheriting https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/v5.4.0/contracts/token/ERC20/extensions/ERC20PermitUpgradeable.sol directly to reuse NoncesKeyed & not have duplicate logic for EIP712